lib/iris/tests/unit/coords/test__discontiguity_in_2d_bounds.py

```diff
@@ -0,0 +1,143 @@
+# (C) British Crown Copyright 2018, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for the `iris.coords._discontiguity_in_2d_bounds` function."""
+
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import numpy as np
+
+from iris.coords import _discontiguity_in_2d_bounds
+
+
+class Test(tests.IrisTest):
+    def setUp(self):
+        self.lon_bounds_3by3 = np.array(
+            [[[0, 1, 1, 0], [1, 2, 2, 1], [2, 3, 3, 2]],
+             [[0, 1, 1, 0], [1, 2, 2, 1], [2, 3, 3, 2]],
+             [[0, 1, 1, 0], [1, 2, 2, 1], [2, 3, 3, 2]]])
+        self.lat_bounds_3by3 = np.array(
+            [[[0, 0, 1, 1], [0, 0, 1, 1], [0, 0, 1, 1]],
+             [[1, 1, 2, 2], [1, 1, 2, 2], [1, 1, 2, 2]],
+             [[2, 2, 3, 3], [2, 2, 3, 3], [2, 2, 3, 3]]])
+        self.discontiguous_bds = np.array(
+            [[[0, 1, 1, 0], [2, 3, 3, 2]],
+             [[1, 2, 2, 1], [2, 3, 3, 2]]])
+
+    def test_bds_wrong_shape_not_3d(self):
+        bds = self.lon_bounds_3by3[0]
+        err_msg = 'Bounds for 2D coordinates must be 3-dimensional and have ' \
+                  '4 bounds per point'
+        with self.assertRaisesRegexp(ValueError, err_msg):
+            _discontiguity_in_2d_bounds(bds)
+
+    def test_bds_wrong_shape_not_4_bounds(self):
+        bds = self.lon_bounds_3by3[:, :, :-2]
+        err_msg = 'Bounds for 2D coordinates must be 3-dimensional and have ' \
+                  '4 bounds per point'
+        with self.assertRaisesRegexp(ValueError, err_msg):
+            _discontiguity_in_2d_bounds(bds)
+
+    def test_contiguous_both_dirs(self):
+        bds = self.lon_bounds_3by3
+        all_eq, diffs_along_x, diffs_along_y = _discontiguity_in_2d_bounds(bds)
+        self.assertTrue(all_eq)
+        self.assertTrue(not diffs_along_x.any())
+        self.assertTrue(not diffs_along_y.any())
+
+    def test_discontiguous_along_x(self):
```

Comment:
A short comment about what you are doing to the bounds array here might be useful.